### PR TITLE
[DA-1731] pid_rid_mapping_api

### PR DESCRIPTION
**Pull request by wangy70**

### CODE REVIEW

**Review by Seeker1911 — APPROVED** [review]
I approve @wangy70 😉 👍

**Comment by wangy70** [comment]
> I approve @wangy70 😉 👍

Wow, come on, submit a PR @Seeker1911

File: rdr_service/dao/participant_dao.py [hunk 1]
```diff
@@ -346,6 +346,38 @@ def get_biobank_ids_sample(self, session, percentage, batch_size):
             .yield_per(batch_size)
         )
 
+    def get_pid_rid_mapping(self, **kwargs):
+        sign_up_after = kwargs.get('sign_up_after')
+        sort = kwargs.get('sort') if kwargs.get('sort') == 'lastModified' else 'signUpTime'
```
Comment (kenny-skaggs) [inline rdr_service/dao/participant_dao.py:351]: Could this potentially be simplified?

File: tests/api_tests/test_participant_api.py [hunk 2]
```diff
@@ -662,6 +662,12 @@ def test_first_ehr_consent_not_set_on_no(self):
         participant_summary = self.send_get("Participant/%s/Summary" % participant_id)
         self.assertNotIn('consentForElectronicHealthRecordsFirstYesAuthored', participant_summary)
 
+    def test_pid_rid_mapping_api(self):
+        self.send_post("Participant", {"providerLink": [self.provider_link_2]})
+        result = self.send_get("ParticipantId/ResearchId/Mapping?signUpAfter=2020-01-01&sort=lastModified")
+        self.assertEqual(len(result['data']), 1)
+        self.assertEqual(result['sort_by'], 'lastModified')
```
Comment (kenny-skaggs) [inline tests/api_tests/test_participant_api.py:671]: What do you think of checking that the response has the research_id field, and that it's correct?

File: rdr_service/dao/participant_dao.py (hunk 1 above)
Comment (wangy70) [reply rdr_service/dao/participant_dao.py:351]: Any suggestion?

Comment (kenny-skaggs) [reply rdr_service/dao/participant_dao.py:351]: Sorry, never mind. I was only looking at this and line 361, I missed that you want to fix it to a specific value to send back on 370 if it wasn't 'lastModified'.

File: tests/api_tests/test_participant_api.py (hunk 2 above)
Comment (wangy70) [reply tests/api_tests/test_participant_api.py:671]: Done

**Review by kenny-skaggs — APPROVED** [review]

**Pull request merged**